### PR TITLE
Ignore multiple components by defining an array of strings

### PR DESCRIPTION
**Pull request by janhoogeveen**
Since my project threw a lot of errors made by things out of my control (DevTools, DockMonitor, IntlProvider and others) I wanted the option to ignore multiple components. This pull request adds that feature, but only for arrays where you define the componentNames as strings. Not sure how to implement regex as well.

### CODE REVIEW

**Comment by davesnx** [comment]
I like your approach, but I really like the approach of having a RegEx. 
Having both, an array of RegEx makes totally sense to me, it's the same as `webpack` is doing and is useful for most use cases.

**Comment by janhoogeveen** [comment]
Absolutely! I don't think it should be too hard to put RegEx in there as well. Maybe the normalize method can also normalize the contents of array values? Then we can always use the .test method in the same way in shouldInclude. I agree this is a nice first step but supporting both should be the goal.

**Comment by janhoogeveen** [comment]
Did a small update so RegEx is supported as well. One downside to this approach is the String comparison (now a regex because of normalisation) will now match even if the displayName is not an exact match of the string.

Example:

```js
{
  exclude: ['Bar']
}
```

will match the component `<FooBar>` as well. I'm not sure how to solve that issue. I suspect this is also an issue for all the other tests as well, not just my contribution.

**Comment by davesnx** [comment]
Cool! @janhoogeveen

File: src/shouldInclude.js [hunk 1]
```diff
@@ -1,8 +1,25 @@
...
+      if (item.test(displayName)) {
+        isExcluded = true;
+        break;
+      }
+    }
+  } else {
+    isExcluded = exclude.test(displayName)
+  }
+
+  isIncluded = include.test(displayName)
```
Comment (garbles) [inline src/shouldInclude.js]: The point of normalizing the options is to remove the need for conditional statements in the rest of the code. 👪

**Comment by garbles** [comment]
Going to close this because of https://github.com/garbles/why-did-you-update/commit/8da002a787a5e0e020c617f5005c3e2e6a898873 and https://github.com/garbles/why-did-you-update/commit/8da002a787a5e0e020c617f5005c3e2e6a898873

**Comment by janhoogeveen** [comment]
@garbles So how would you exclude multiple components now? Use the pipe operator in a regex string?

**Comment by garbles** [comment]
@janhoogeveen you can supply an array of either strings or regexps. See this https://github.com/garbles/why-did-you-update/blob/master/tests/index-test.js#L214-L243

**Comment by garbles** [comment]
My mistake. The above comment included the wrong commit. https://github.com/garbles/why-did-you-update/commit/418d589480e875897cccb3d1711cda8b0f86d42b

**Comment by garbles** [comment]
😅 😅 😅 😅 😅 😅 😅 😅 😅 😅 😅 😅 😅

**Comment by janhoogeveen** [comment]
I see, looks great! Much cleaner this way. Thanks!

**Pull request closed without merging**